packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -346,6 +346,22 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
             }
         }
 
+        /**
+         * If the request has "errorOnResourceDeleted" header, we need to return an error if the data store being
+         * requested has been deleted.
+         *
+         * This is used to handle scenario where a data store shared with an external app is deleted and hence GC'd.
+         * The error returned signals the app to take appropriate action for deleted content.
+         */
+        if (request.headers?.errorOnResourceDeleted === true) {
```

Comment:
let's be more explicit with the header name, maybe something like errorOnUnreferencedResources